src/Adyen/HttpClient/ClientInterface.php

```diff
@@ -33,4 +33,19 @@ public function requestPost(\Adyen\Service $service, $requestUrl, $params);
      * @return mixed
      */
     public function requestHttp(\Adyen\Service $service, $requestUrl, $params, $method, $requestOptions = null);
+
+    /**
+     * @param \Adyen\Service $service
+     * @param $requestUrl
+     * @param $bodyParams
+     * @param $method
+     * @return mixed
+     */
+    public function requestHttpRest(
```

Comment:
Why does this service need a new public interface method? Can't it rely on existing features of the library like Management API did?

Reply:
Because of this line:  https://github.com/Adyen/adyen-php-api-library/blob/bcb1198ebce18e20abc94adcc16bb31c613dc293/src/Adyen/HttpClient/CurlClient.php#L369

Basically this prevents us from having queryparams for Post/Patch and having request body for delete for example. So I just opted for another function that splits the query params from the bodyparams.